### PR TITLE
Support Docker ServiceDiscovery

### PR DESCRIPTION
**Pull request by lujiajing1126**
In this PR, I've added support for docker service discovery as [documented in Prometheus configuration section](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#docker_sd_config)

The `docker` and `dockerSwarm` have been merged since they are actually using the same API(s)

### CODE REVIEW

File: lib/promscrape/discovery/docker/containers.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	if err := json.Unmarshal(data, &containers); err != nil {
+		return nil, fmt.Errorf("cannot parse containers: %w", err)
+	}
+	return containers, nil
+}
+
+func addContainersLabels(containers []container, networkLabels map[string]map[string]string, defaultPort int) []map[string]string {
+	var ms []map[string]string
+	for _, c := range containers {
+		commonLabels := map[string]string{
```
Comment (f41gh7) [inline lib/promscrape/discovery/docker/containers.go:66]: I suggest to check for c.Names len, probably its not initialized. 

```go
		if len(c.Names) == 0 {
			continue
		}
```

**Review by f41gh7 — COMMENTED** [review]
Looks good to me. Have only one comment. 

 Thanks for refactoring!

File: lib/promscrape/discovery/docker/containers.go (hunk 1 above)
Comment (lujiajing1126) [reply lib/promscrape/discovery/docker/containers.go:70]: Added

**Review by valyala — APPROVED** [review]
LGTM

**Comment by valyala** [comment]
@lujiajing1126 , thanks for the contribution!

**Comment by valyala** [comment]
The `docker_sd_config` service discovery has been added to [v1.62.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Pull request merged**